src/geometry/detail/VGNavCollection.hh

```diff
@@ -0,0 +1,204 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file VGNavCollection.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#ifndef __CUDA_ARCH__
+#    include <memory>
+#endif
+#include <VecGeom/navigation/NavigationState.h>
+#include "base/Assert.hh"
+#include "base/OpaqueId.hh"
+#include "base/Types.hh"
+
+//---------------------------------------------------------------------------//
+/* VECGEOM FORWARD DECLARATIONS
+ *
+ * CUDA declarations aren't reachable from C++ host-compiled code, and the
+ * reverse, so we must fudge by forward-declaring here as though
+ * NavigationState is an untemplated class, not a type alias, for each unused
+ * memspace.
+ */
+namespace vecgeom
+{
+#ifndef __NVCC__
+inline
+#endif
+    namespace cxx
+{
+class NavStatePool;
+} // namespace cxx
+} // namespace vecgeom
+namespace vecgeom
+{
+#ifdef __NVCC__
+namespace cxx
+#else
+namespace cuda
+#endif
+{
+class NavigationState;
+}
```

Comment:
This seems to stems from omissions in the VecGeom header files.  I will take a look.